### ISSUE TITLE
BUG: Use C99 flexible arrays instead of type[1] trailing member

### ISSUE THREAD

**Issue by Flamefire** [erikd/libsamplerate#77]
Related to #68 as it requires C99 (although C99 features are already used)

C99 flexible arrays are used at some points, e.g.: https://github.com/erikd/libsamplerate/blob/2647fd9bac5f11f24b5f1ee55806671f81520f66/src/src_sinc.c#L57

At other the idiom of having a trailing `float[1]` member is used: https://github.com/erikd/libsamplerate/blob/2647fd9bac5f11f24b5f1ee55806671f81520f66/src/src_linear.c#L34

A SO answer highly recommends using the C99 feature: https://stackoverflow.com/a/247040/1930508 with a comment that using `float[1]` may produce incorrect code: https://lkml.org/lkml/2015/2/18/407

Furthermore to much memory is allocated (1 additional float): https://github.com/erikd/libsamplerate/blob/2647fd9bac5f11f24b5f1ee55806671f81520f66/src/src_linear.c#L174

Bottom line: 
- Require C99 compiler
- Use flexible arrays instead of trailing `float[1]`
